runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchPortablePipelineTranslator.java

```diff
@@ -254,15 +270,22 @@ public void translate(
             stagePayload.getEnvironment(),
             PipelineOptionsTranslation.toProto(context.getPipelineOptions()),
             outputMap);
-    Map<String, String> inputs = transform.getInputsMap();
-    DataSet<WindowedValue<InputT>> inputDataSet =
-        context.getDataSetOrThrow(Iterables.getOnlyElement(inputs.values()));
-    DataSet<RawUnionValue> taggedDataset =
+
+    DataSet<WindowedValue<InputT>> inputDataSet = context.getDataSetOrThrow(stagePayload.getInput());
+
+    MapPartitionOperator<WindowedValue<InputT>, RawUnionValue> taggedDataset =
```

Comment:
Does this actually change the semantics of the Flink pipeline itself? What's the difference?